avaframe/com1DFA/deriveParameterSet.py

```diff
@@ -197,3 +194,245 @@ def checkRelEntThVariation(cfg, variationDict):
         if key in variationDict and cfg['GENERAL'].getboolean(flag) is False:
             log.warning('Parameter variation for %s thickness not working as %s read from shp file - \
                          consider setting %s to True' % (value, key, flag))
+
+
+def getThicknessValue(cfg, inputSimFiles, fName, thType):
+    """ set thickness values according to settings chosen and add info to cfg
+        if thFromShp = True - add in INPUT section thickness and id info
+        if thFromShp = False - check format of thickness value in GENERAL section
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings
+        inputSimFiles: dict
+            dictionary with info on input files and attributes (id and thickness)
+        fName: str
+            name of scenario shp file (entrainment, release, ...)
+        thType: str
+            thickness parameter name (entTh, relTh, ...)
+
+        Returns
+        -------
+        cfg: configparser object
+            updated configuration settings with info on actual thickness values to be used for simulations
+
+        """
+
+    # fetch thickness values from shapefile
+    thicknessList = inputSimFiles[fName]['thickness']
+    idList = inputSimFiles[fName]['id']
+
+    # create key names for flags and parameter variation info
+    thFlag = thType + 'FromShp'
+    thPercent = thType + 'PercentVariation'
+
+    # if thickness should be read from shape file
+    if cfg['GENERAL'].getboolean(thFlag):
+        # if at least one but not all features in a shapefile have a thickness value - error
+        if ('None' in thicknessList):
+            message = 'Not all features in shape file have a thickness value - check shape file attributes: %s' % fName
+            log.error(message)
+            raise AssertionError(message)
+        else:
+            # set thickness value in ini file from info of shape file
+            thId = idList[0]
+            thThickness = thicknessList[0]
+            for count, id in enumerate(idList[1:]):
+                thId = thId +'|' + id
+                thThickness = thThickness + '|'+ thicknessList[count+1]
+
+            # add in INPUT section
+            cfg['INPUT'][thType + 'Id'] = thId
+            cfg['INPUT'][thType + 'Thickness'] = thThickness
+
+    else:
+        # if thickness should be read from ini file - check if format is correct
+        if '$' in cfg['GENERAL'][thType] and len(cfg['GENERAL'][thType].split('$')) != 3:
+            message = 'Format of relTh value in ini file is not correct - for variation from ini use refValue$percent$numberOfSteps'
+            log.error(message)
+            raise AssertionError(message)
+
+    return cfg
+
+
+def checkThicknessSettings(cfg, thName):
```

Comment:
Maybe return a boolean like `thicknessIsCorrect` os similar so it can be used in the calling function

Reply:
added a flag thicknessSettingsCorrect